src/Roslyn.Suppressions/Roslyn.Suppressions/SettingsFile/RoslynSettingsFileStorage.cs

```diff
@@ -87,15 +90,16 @@ public RoslynSettings Get(string settingsKey)
             return null;
         }
 
-        public void Update(RoslynSettings settings)
+        public void Update(RoslynSettings settings, string solutionNameWithoutExtension)
         {
             Debug.Assert(settings != null, "Not expecting settings to be null");
             Debug.Assert(!string.IsNullOrWhiteSpace(settings.SonarProjectKey), "Not expecting settings.SonarProjectKey to be null");
+            Debug.Assert(solutionNameWithoutExtension != null, "Not expecting solutionNameWithoutExtension to be null");
 
             try
             {
                 CodeMarkers.Instance.FileStorageUpdateStart();
-                var filePath = RoslynSettingsFileInfo.GetSettingsFilePath(settings.SonarProjectKey);
+                var filePath = RoslynSettingsFileInfo.GetSettingsFilePath(solutionNameWithoutExtension);
```

Comment:
Previously the settings file path contained the Sonar project key; now it contains the solution name - see #4188.

I forgot it would impact the Roslyn suppressions. Kudos to @georgii-borovinskikh-sonarsource  for spotting the problem.